integration/simulation/transaction_injector.go

```diff
@@ -311,21 +318,6 @@ func (m *TransactionInjector) rndL2NodeClient() *obscuroclient.Client {
 	return m.l2Clients[rand.Intn(len(m.l2Clients))] //nolint:gosec
 }
 
-func newObscuroTransferTx(from wallet.Wallet, dest common.Address, amount uint64, client *obscuroclient.Client) types.TxData {
```

Comment:
why did you remove these functions?
I think they were adding clarity to what's going on

Reply:
I don't really agree.. They seem to be doing one instruction, the tx data generation.

And it's starting to have super similar instructions like `newCustomObscuroWithdrawalTx` and `newObscuroWithdrawalTx` which is only one more step to follow, to figure out what it's doing.

Having to make then a part of the struct feels super bloated.